Find a functionally equivalent code:android.security.KeyChain.choosePrivateKeyAlias:COMMENT
Method Modifier: public      static      
Comment:/**
 * Launches an {@code Activity} for the user to select the alias
 * for a private key and certificate pair for authentication. The
 * selected alias or null will be returned via the
 * KeyChainAliasCallback callback.
 *
 * <p>{@code keyTypes} and {@code issuers} may be used to
 * highlight suggested choices to the user, although to cope with
 * sometimes erroneous values provided by servers, the user may be
 * able to override these suggestions.
 *
 * <p>{@code host} and {@code port} may be used to give the user
 * more context about the server requesting the credentials.
 *
 * <p>{@code alias} allows the chooser to preselect an existing
 * alias which will still be subject to user confirmation.
 *
 * @param activity The {@link Activity} context to use for
 * launching the new sub-Activity to prompt the user to select
 * a private key; used only to call startActivity(); must not
 * be null.
 * @param response Callback to invoke when the request completes;
 * must not be null
 * @param keyTypes The acceptable types of asymmetric keys such as
 * "RSA" or "DSA", or a null array.
 * @param issuers The acceptable certificate issuers for the
 * certificate matching the private key, or null.
 * @param host The host name of the server requesting the
 * certificate, or null if unavailable.
 * @param port The port number of the server requesting the
 * certificate, or -1 if unavailable.
 * @param alias The alias to preselect if available, or null if
 * unavailable.
 */

@@ -22,27 +22,28 @@
         throw new NullPointerException("response == null");
     }
     Intent intent = new Intent(ACTION_CHOOSER);
+    intent.setPackage(KEYCHAIN_PACKAGE);
     intent.putExtra(EXTRA_RESPONSE, new AliasResponse(response));
     intent.putExtra(EXTRA_HOST, host);
     intent.putExtra(EXTRA_PORT, port);

